src/mongo/MongoScrapbook.ts

```diff
@@ -131,8 +136,23 @@ async function executeCommand(context: IActionContext, command: MongoCommand): P
                 const docNode = new MongoDocumentTreeItem(colNode, document);
                 await ext.fileSystem.showTextDocument(docNode, { viewColumn: vscode.ViewColumn.Beside });
             } else {
-                const viewColumn = vscode.window.activeTextEditor?.viewColumn;
-                await vscodeUtil.showNewFile(result, 'result', '.json', viewColumn ? viewColumn + 1 : undefined);
+                const showOptions: vscode.TextDocumentShowOptions = { viewColumn: vscode.ViewColumn.Beside };
+                readOnlyContent = readOnlyContent || defaultReadOnlyContent;
+                result += `${EOL}${EOL}`;
+
+                if (readOnlyContent) {
+                    await readOnlyContent.append(result);
+
+                    // If the read only content isn't visible anymore, re-show it. Appending to a hidden document does not show it
+                    if (!(await readOnlyContent.isVisible())) {
```

Comment:
Should be able to get rid of this, too, since `openReadOnlyContent` was called at the beginning of "Execute All". If users click away from the output _while_ the commands are running, I feel like that's on them and we would just be annoying if we re-show it

Reply:
The main reason I added that check is because if the output is closed, it won't be re-opened on its own when appending. So the results wouldn't be displayed in the case where `executeCommand` is called without `readOnlyContent`.

Should I not re-show in the "Execute All" case?

Reply:
Per offline discussion: I misinterpreted part of the original issue, we don't want to append content to the output if a single command is run